### PR TITLE
fix(dev): don't overwrite proxy headers if already set

### LINKED ISSUE #894: allow overriding `x-forwarded-for` header in development mode
<table><td>

&nbsp;&nbsp;&nbsp;Forwarded from downstream issue:
- https://github.com/nuxt/nuxt/issues/18609 by @Irelynx

</td></table><br>

### Environment

- Operating System: `Windows_NT`
- Node Version:     `v18.6.0`
- Nuxt Version:     `3.1.1`
- Nitro Version:    `2.1.0`
- Package Manager:  `yarn@1.22.19`
- Builder:          `vite`
- User Config:      `typescript`, `runtimeConfig`, `modules`, `content`
- Runtime Modules:  `@nuxt/content@2.4.2`
- Build Modules:    `-`

### Reproduction

Can't reproduce on StackBlitz or CodeSandbox, so here is the code:

```typescript
// file: /server/middleware/log.ts
export default defineEventHandler((event) => {
  console.log(
    'New request:',
    event.node.req.url,
    'from:',
    {
      cRemoteAddress: event.node.req.connection.remoteAddress,
      sRemoteAddress: event.node.req.socket.remoteAddress,
      hForwardedFor: event.node.req.headers['x-forwarded-for'],
    }
  );
});
```
```typescript
// file: /server/api/count.ts
export default defineEventHandler((event) => {
  return { hello: 'world' };
});
```
---
```javascript
// file: /node.server.js
const http = require('http');

const server = http.createServer((req, res) => {
  console.log('native node server', {
    cRemoteAddress: req.connection.remoteAddress,
    sRemoteAddress: req.socket.remoteAddress,
    hForwardedFor: req.headers['x-forwarded-for'],
  });
  res.end();
});
server.listen(3000, () => {
  console.log('server is listening');
});
```

### Describe the bug

I was trying to get Client IP address behind Nginx proxy.
The main issue is:
`event.node.req.headers['x-forwarded-for']` was rewritten by nuxt (or other dependency, i can't figure out), so it is not possible to determine original request IP address.

Reproduction flow:
Client (Global IP: 1.2.3.4) -> Nginx proxy (local IP: 192.168.40.4) -> Nuxt server

* Nuxt middleware (`/server/middleware/log.ts`) output:
  ```bash
  New request: /api/counter from: {
    cRemoteAddress: undefined, # event.node.req.connection.remoteAddress
    sRemoteAddress: undefined, # event.node.req.socket.remoteAddress
    hForwardedFor: '::ffff:192.168.40.4' # rewritten by something! event.node.req.headers['x-forwarded-for']
  }
  ```
* Node.js http server (`/node.server.js`) output:
  ```bash
  native node server {
    cRemoteAddress: '::ffff:192.168.40.4', # req.connection.remoteAddress
    sRemoteAddress: '::ffff:192.168.40.4', # req.socket.remoteAddress
    hForwardedFor: '1.2.3.4' # real header value from Nginx, req.headers['x-forwarded-for']
  }
  ```

### Additional context

_No response_

### Logs

_No response_

### PR DESCRIPTION
**Pull request by MikeBellika**
### 🔗 Linked issue
#894 

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Looks at `X-Forwarded` headers in `proxyReq` and doesn't overwrite them if they're present.

Resolves #894
### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/dev/server.ts [hunk 1]
```diff
@@ -158,14 +158,16 @@ export function createDevServer(nitro: Nitro): NitroDevServer {
   // Main worker proxy
   const proxy = createProxy();
   proxy.proxy.on("proxyReq", (proxyReq, req) => {
-    // TODO: Avoid overwriting headers if already set
-    if (req.socket.remoteAddress) {
+    if (req.socket.remoteAddress && !proxyReq.getHeaders()["x-forwarded-for"]) {
       proxyReq.setHeader("X-Forwarded-For", req.socket.remoteAddress);
     }
-    if (req.socket.remotePort) {
+    if (req.socket.remotePort && !proxyReq.getHeaders()["x-forwarded-port"]) {
```
Comment (pi0) [inline src/dev/server.ts:164]: Perf: We don't need to call getHeaders twice. Instead, we can keep it in a variable before L160

Comment (MikeBellika) [reply src/dev/server.ts:164]: I'll fix

Comment (MikeBellika) [reply src/dev/server.ts]: I have committed a fix. I don't think we can do it before L160, since we don't have `proxyReq` available

**Review by pi0 — APPROVED** [review]
Thanks!

File: src/dev/server.ts (hunk 1 above)
Comment (pi0) [reply src/dev/server.ts]: That's a right place

**Pull request merged**